arb_os/inbox.mini

```diff
@@ -157,24 +157,57 @@ public impure func inbox_get() -> IncomingRequest {
         if (ebMsg.blockNumber > globalInbox.blockNum) {
             if ((globalInbox.sequencer != None<SequencerState>) && globalInbox.seenMsgAtThisBlockNum) {
-                let eobMsg = endOfBlockMessage(globalInbox.blockNum, globalInbox.timestamp);
-                globalInbox = globalInbox with {
-                    blockNum: ebMsg.blockNumber
-                } with {
-                    timestamp: ebMsg.timestamp
-                } with {
-                    seenMsgAtThisBlockNum: false
-                };
-                return eobMsg;
+                // don't update queue, because we're not consuming ebMsg
+                let rawSideloadMsg = asm(globalInbox.blockNum,) any { sideload };
+                if (rawSideloadMsg == ()) {
+                    let eobMsg = endOfBlockMessage(globalInbox.blockNum, globalInbox.timestamp);
+                    globalInbox = globalInbox with {
+                        blockNum: ebMsg.blockNumber
+                    } with {
+                        timestamp: ebMsg.timestamp
+                    } with {
+                        seenMsgAtThisBlockNum: false
+                    };
+                    return eobMsg;
+                } else {
+                    let newMsgRaw = unsafecast<IncomingRequestFromInbox>(rawSideloadMsg);
+                    let newMsg = struct {
+                        kind: newMsgRaw.kind,
+                        blockNumber: newMsgRaw.blockNumber,
+                        timestamp: newMsgRaw.timestamp,
+                        sender: newMsgRaw.sender,
+                        requestId: newMsgRaw.requestId,
+                        msgData: bytearray_fromSizeAndBuffer(newMsgRaw.msgSize, newMsgRaw.msgData),
+                        provenance: struct {
+                            l1SeqNum: newMsgRaw.requestId,
+                            parentRequestId: 0,
+                            indexInParent: ~0,
+                        },
+                        aggregator: None<AggregatorInfo>,
+                        adminMode: true,
+                    };
+
+                    if (newMsg.blockNumber == 0) {
+                        newMsg = newMsg with { blockNumber: globalInbox.blockNum };
+                    }
+                    if (newMsg.timestamp == 0) {
+                        newMsg = newMsg with { timestamp: globalInbox.timestamp };
+                    }
+                    return newMsg;
+                }
             } else {
                 globalInbox = globalInbox with {
+                    queue: updatedQ
+                } with {
                     blockNum: ebMsg.blockNumber
                 } with {
                     timestamp: ebMsg.timestamp
                 } with {
                     seenMsgAtThisBlockNum: true
                 };
             }
+        } else {
+            globalInbox = globalInbox with { queue: updatedQ };
         }
 
         globalInbox = globalInbox with { queue: updatedQ };
```

Comment:
It seems like the `globalInbox` is being updated twice in all the code paths that reach this point. Maybe it can just be updated here and not in the other places?

Reply:
Removed the redundant assignment.